storage/src/merkle_storage.rs

```diff
@@ -1163,11 +1174,10 @@ impl MerkleStorage {
     fn get_entries_recursively(
         &self,
         entry: &Entry,
-        batch: &mut WriteBatch,
+        batch: &mut Vec<(EntryHash, ContextValue)>,
     ) -> Result<(), MerkleError> {
         // add entry to batch
-        self.db
-            .put_batch(batch, &self.hash_entry(entry)?, &bincode::serialize(entry)?)?;
```

Comment:
Can this cause problems? I would guess that the RocksDB implementation for this is optimized, while the new approach isn't (at least not as much).

Reply:
i have already exteneded trait and made use of write batch :+1: